common/src/main/java/io/druid/common/guava/GuavaUtils.java

```diff
@@ -62,4 +66,9 @@ public static Long tryParseLong(@Nullable String string)
 
     return null;
   }
+
+  public static <V> ListenableFuture<List<V>> allFuturesAsList(Stream<ListenableFuture<? extends V>> futures)
+  {
+    return Futures.allAsList(futures::iterator);
```

Comment:
It doesn't win anything from the Stream's laziness perspective, because Futures.allAsList() copies all futures into a `List` anyway.

Reply:
That's fine. I'll add a java doc to clarify.

Reply:
added some docs